Samples/DrTestActionSampleVM/DrTestActionSampleVM/WrapperDrVM.cs

```diff
@@ -258,6 +271,339 @@ protected virtual VirtualMachine GetVirtualMachine(string vmName)
             return (T)vClient.FindEntityView(typeof(T), null, filter, null);
         }
 
+
+
+        /// <summary>
+        /// 
+        /// </summary>
+        /// <param name="vm"></param>
+        /// <param name="snapshotName"></param>
+        /// <returns></returns>
+        private static ManagedObjectReference HelperSnapshotChecker(VirtualMachineSnapshotTree[] rootSnapshotList, string snapshotName, VirtualMachine vm)
+        {
+
+
+
+            ManagedObjectReference resultInfo = new ManagedObjectReference();
+            int count = 0;
+            if (rootSnapshotList != null)
+            {
+                VirtualMachineSnapshotTree snapshot = rootSnapshotList[0];
+                for (int i = 0; i < rootSnapshotList.Length; i++)
+                {
+                    snapshot = rootSnapshotList[i];
+                    String name = snapshot.Name;
+                    if (name == snapshotName)
+                    {
+                        resultInfo = snapshot.Snapshot;
+                        count++;
+                        if (count > 1) throw new VMSnapshotHaveMoreThenOneExeption(vm.Name, snapshotName);
+                    }
+                    if (rootSnapshotList[i].ChildSnapshotList != null)
+                    {
+                        var tempm = HelperSnapshotChecker(rootSnapshotList[i].ChildSnapshotList, snapshotName, vm);
+                        if (tempm.Value != null)
+                        {
+                            resultInfo = tempm;
+                            count++;
+                            if (count > 1) throw new VMSnapshotHaveMoreThenOneExeption(vm.Name, snapshotName);
+                        }
+                    }
+                }
+            }
+            return resultInfo;
+        }
+
+
+        /// <summary>
+        /// 
+        /// </summary>
+        /// <param name="vm"></param>
+        /// <param name="host"></param>
+        /// <returns></returns>
+        private static bool HelperCheckerIncludeHost(VirtualMachine vm, HostSystem host)
+        {
+            foreach (var VMHost in host.Vm)
+            {
+                if (VMHost.ToString() == vm.MoRef.ToString())
+                {
+                    return true;
+                }
+            }
+            return false;
+        }
+
+
+        /// <summary>
+        /// 
+        /// </summary>
+        /// <param name="vmName"></param>
+        /// <param name="newName"></param>
+        internal virtual void CloneVM(string vmName, string newName, string hostName, string snapshotName)
```

Comment:
should add summary. should support target host name. need to rename newName -> newVmName